### ISSUE TITLE
Rollback to previous state

### ISSUE THREAD

**Issue by jantrienes** [typicode/json-server#115]
It would be handy, to restore the DB to a previous state. Either to one of the created snapshots or to the initial state. Especially for demonstrations, where you like to undo certain operations.

**Comment by syzer** [comment]
:+1:

**Comment by ftheomunhoz** [comment]
I would be useful if there was an option to start the server as "readonly" also.

**Comment by syzer** [comment]
so you would reject all POST/PUT?

**Comment by ftheomunhoz** [comment]
If configured for it, yeah. Something else we found pretty useful was to configure a map to reply a specific response message, given a request. That also allows to mock error messages and such.

**Comment by syzer** [comment]
you can put proxy before json-server to catch all PUT/POST and send 403 
(10 LOC)

**Comment by ftheomunhoz** [comment]
Yeah but that would require another component to handle,start, maintain...

**Comment by syzer** [comment]
come on it's: 10 LOC.

**Comment by ftheomunhoz** [comment]
I was thinking about something like this https://github.com/atecarlos/protractor-http-mock

**Comment by typicode** [comment]
@ftheomunhoz for the moment, to disable some verbs, you can use the project as a module and add this:

```js
var server = jsonServer.create()

// Filter by req.method
server.all('*', function (req, res, next) {
  if (req.method === 'GET') {
    next() // Continue
  } else {
    res.sendStatus(403) // Forbidden
  }
})
```

Since there's quite a lot of requests for a `read-only` mode, I'll consider adding it as an option too.

**Comment by ftheomunhoz** [comment]
@typicode Thanks! I will follow with your suggestion for now. @americoneto1 you might be interested in this as well.

**Comment by americoneto1** [comment]
I also appreciate this option!

**Comment by typicode** [comment]
@ftheomunhoz @americoneto1 it's available in `v0.8.6`. Thank you for the feedback :+1:

**Comment by americoneto1** [comment]
Thank you @typicode ! I'll test this option
